### PR TITLE
feat: play sound on successful swaps

### LINKED ISSUE #338: UX: Add swap success sound
compare mempool.space's iconic sounds

### PR DESCRIPTION
**Pull request by dni**
not very happy with the sound still, hard to find a good one

closes #338

screenshot-1716362712

### CODE REVIEW

**Comment by kilrau** [comment]
Nice, but it should play independently from notifications, just like mempool.space

**Comment by kilrau** [comment]
Don't have "Play sound". Just play it when one enables it via the toggle.

Also, please try a toggle style without writing "on/off" out. Or at least only one of them. These toggles were invented to signify on/off...

Also on right, off left seems to be standard these days

[image1da0-4a7d-a8a7-cf12d79d01db)
](https://www.google.com/search?q=on+off+toggle+switch+phone&tbm=isch&ved=2ahUKEwjSgMn65viFAxU9U6QEHehIDW0Q2-cCegQIABAA&oq=on+off+toggle+switch+phone&gs_lp=EgNpbWciGm9uIG9mZiB0b2dnbGUgc3dpdGNoIHBob25lSO0RULkGWMIOcAB4AJABAJgBUKABiQSqAQE3uAEDyAEA-AEBigILZ3dzLXdpei1pbWfCAgoQABiABBhDGIoFwgIFEAAYgATCAgYQABgHGB7CAgQQABgewgIGEAAYCBgewgIHEAAYgAQYGIgGAQ&sclient=img&ei=Mq84ZtKfNr2mkdUP6JG16AY&bih=1578&biw=1536&client=ubuntu-chr&hs=THf)

File: src/components/Audio.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useGlobalContext } from "../context/Global";
+
+const Audio = () => {
```
Comment (michael1011) [inline src/components/Audio.tsx:3]:
```suggestion
const Audio = () => {
```

Give this a proper name. Like `SwapSuccessSoundSetting`

File: src/components/RefundButton.tsx [hunk 2]
```diff
@@ -149,6 +151,7 @@ const RefundButton = ({
                     setRefundTxId(res.refundTx);
                 }
             }
+            if (audio()) playSound();
```
Comment (michael1011) [inline src/components/RefundButton.tsx:154]: I don't think that we should play a sound on refund

File: src/context/Global.tsx [hunk 3]
```diff
@@ -146,6 +149,7 @@ const GlobalProvider = (props: { children: any }) => {
     });
 
     const [settingsMenu, setSettingsMenu] = createSignal<boolean>(false);
+    const [audio, setAudio] = createSignal<boolean>(false);
```
Comment (michael1011) [inline src/context/Global.tsx:152]:
```suggestion
    const [swapSuccessSound, setSwapSuccessSound = createSignal<boolean>(false);
```

This should be a persistent signal

File: src/context/Global.tsx [hunk 4]
```diff
@@ -160,6 +164,12 @@ const GlobalProvider = (props: { children: any }) => {
         setNotification(message);
     };
 
+    const playSound = () => {
```
Comment (michael1011) [inline src/context/Global.tsx:167]:
```suggestion
    const swapSuccessSound = () => {
```

File: src/context/Global.tsx [hunk 5]
```diff
@@ -160,6 +164,12 @@ const GlobalProvider = (props: { children: any }) => {
...
+        const audio = new Audio("/sound.mp3");
```
Comment (michael1011) [inline src/context/Global.tsx:168]: Give the mp3 a proper file name too. Like eg `swapSuccessSound.mp3`

File: src/i18n/i18n.ts [hunk 6]
```diff
@@ -181,6 +181,10 @@ const dict = {
         denomination_tooltip: "Choose your preferred denomination: BTC or sats",
         decimal_tooltip:
             "Choose your preferred decimal separator: dot or comma",
+        enable_audio: "Enable Audio",
```
Comment (michael1011) [inline src/i18n/i18n.ts:184]: @kilrau check translations

**Review by michael1011 — CHANGES_REQUESTED** [review]

File: src/context/Global.tsx [hunk 7]
```diff
@@ -152,6 +155,8 @@ const GlobalProvider = (props: { children: any }) => {
     });
 
     const [settingsMenu, setSettingsMenu] = createSignal<boolean>(false);
+    const [audioNotification, setAudioNotification] =
+        createSignal<boolean>(false);
```
Comment (michael1011) [inline src/context/Global.tsx:159]: This needs to be persistent in local storage

File: src/context/Global.tsx [hunk 8]
```diff
@@ -161,9 +166,16 @@ const GlobalProvider = (props: { children: any }) => {
         },
     );
 
-    const notify = (type: string, message: string) => {
+    const notify = (type: string, message: string, audio?: boolean) => {
```
Comment (michael1011) [inline src/context/Global.tsx:169]:
```suggestion
    const notify = (type: string, message: string, audio: boolean = false) => {
```

**Review by michael1011 — CHANGES_REQUESTED** [review]

**Comment by kilrau** [comment]
translations look good

**Review by michael1011 — APPROVED** [review]

**Pull request merged**